### DISCUSSION TITLE
Matter configuration Tasmota v. 13.3.0 Quinlan

### DISCUSSION THREAD

**Discussion by Bebuccio** [arendst/Tasmota#20231] [Q&A]
Good morning.
I am trying to configure an ESP32 (tasmota ver. 13.3.0) with the Matter configuration
I have the ESP32 configured with one Relay and one TemperatureSensor (internal cpu temperature).
Enabling matter autodiscover the devices found are:
# | Name | Type | Parameter |  
1 |   | Relay |   | 
2 |   | Temperature |   | 

Then I have configured the ESP32 to work as matter bridge setting the IP address of another ES8266 device with a relay and a temperature sensor (AM2301)

This is the matter configuration of the other devices:
SOnOff-Divano
-- | -- | -- | -- | --
60 |   | Light 0 On |   | 
61 |   | Temperature |   | 
62 |   | Humidity |   | 

This is the console-log of ESP32:
00:00:03.556 WIF: Connected
00:00:03.634 MTR: Configuring endpoints
00:00:03.637 MTR:   endpoint =     0 type:root
00:00:03.644 MTR:   endpoint =     1 type:relay relay:1
00:00:03.651 MTR:   endpoint =     2 type:temperature filter:ESP32#Temperature
00:00:03.652 MTR:   endpoint = 65280 type:aggregator
00:00:03.654 RSL: RESULT = {"Matter":{"Initialized":1}}
00:00:04.114 MTR: =Saved     parameters and configuration
00:00:04.120 MTR: Starting UDP server on port: 5540

18:29:20.651 MTR: adding endpoint = 60 type:http_light0 relay:1 url:xxx.xxx.xxx.xxx
18:29:20.689 MTR: =Saved     parameters and configuration
18:29:20.712 MTR: adding endpoint = 61 type:http_temperature filter:AM2301#Temperature url:xxx.xxx.xxx.xxx
18:29:20.756 MTR: =Saved     parameters and configuration
18:29:20.775 MTR: adding endpoint = 62 type:http_humidity filter:AM2301#Humidity url:xxx.xxx.xxx.xxx
18:29:20.827 MTR: =Saved     parameters and configuration

Then I have performed the matter-discovery telling my Alexa-Dot to look for.
This is the console-log

18:29:23.221 MTR: Manual pairing code: yyyyyyy
18:29:23.229 RSL: RESULT = {"Matter":{"Commissioning":1,"PairingCode":"yyyyyyyyyyyyyyy","QRCode":"xxxxxxxxxxxxxxxxxxx"}}
18:29:24.819 MTR: =Saved     parameters and configuration
18:29:25.884 MTR: =Saved     parameters and configuration
18:29:29.147 MTR: =Saved     parameters and configuration
18:30:10.379 MTR: New Commissioning (PASE id=30001) from [192.168.20.204]:5541
18:30:14.135 RSL: RESULT = {"Matter":{"Commissioning":0}}
18:30:15.838 MTR: New Connection (CASE id=51467) from [192.168.20.204]:5541
18:30:16.765 MTR: =Saved     2 fabric(s) and 6 session(s)
18:30:18.108 MTR: =Saved     3 fabric(s) and 7 session(s)
18:30:18.146 MTR: --- Commissioning complete for Fabric 'yyyyyyyyyy' (Vendor Amazon Alexa) ---
18:31:03.609 MTR: New Connection (CASE id=663) from [192.168.20.205]:5541
18:31:04.809 MTR: =Saved     3 fabric(s) and 8 session(s)

The result on the Alexa device is that when I look for the temperature sensor AM2301 it report "-1.0" value

Performing the command MTRINFO on the console, I see this:

18:32:10.785 CMD: mtrinfo
18:32:10.799 RSL: RESULT = {"MtrInfo":{"Ep":1,"Power":0}}
18:32:10.808 RSL: RESULT = {"MtrInfo":{"Ep":2,"Temperature":5330}}
18:32:10.820 RSL: RESULT = {"MtrInfo":{"Ep":60,"Power":0}}
18:32:10.836 RSL: RESULT = {"MtrInfo":"Done"}

!! The device #61 was disappeared

If I configure the ESP32 with only the temperature-sensor (internal cpu temp without the relay-port) and the bridge with the same ESP8266 device (relay and AM2301), Alexa can see all the devices (temperature-esp32, relay-ES8266 and AM2301-ES8266)

This problem was present also in the previous 13.2.0 version

Anothe question: it is possible to reset the numeration of the bridged-devices? The number-ID of the internal ESP32 devices starts by 1. The bridged devices now have ID 60, 61 and 62. If I reset the configurationeand rescan for bridged devices, the ID will start from 63 (next to the previous last ID)

**Comment by s-hadinger** [comment]
The Matter standard forbids from reusing endpoint numbers once they have been reported to the controller. In practice, you can edit the json file `_matter_device.json` and change manually the endpoint numbers to suit your needs.

I never tried a temperature sensor with Alexa, this must have been a recent addition. For troubleshooting, I need to see the logs of Tasmota reporting the temperature. It requires `weblog 3` or preferably `seriallog 3` if you have a serial terminal attached to it

**Reply by Bebuccio** [reply]
The problem is that if I configure ESP32 ONLY with a temp-sensor, with MRTINFO command I see the correct deviceId=1 (temp):
14:08:29.478 CMD: mtrinfo
14:08:29.489 RSL: RESULT = {"MtrInfo":{"Ep":1,"Temperature":5330}}
14:08:29.495 RSL: RESULT = {"MtrInfo":"Done"}
Adding, on bridge-mode, the esp8266 with a Relay, AM2301-Temp, AM2301-Humid, the display of matter devices (using MTRINFO command) show ONLY the previous ESP32-temp-device and the new-one ES8266-relay (MTRINFO don't show the third device (AM2301-temp):
14:08:42.201 MTR: adding endpoint = 2 type:http_light0 relay:1 url:xxx.yyy.www.zzz
14:08:42.247 MTR: =Saved     parameters and configuration
14:08:42.271 MTR: **adding endpoint = 3 type:http_temperature filter:AM2301#Temperature** url:xxx.yyy.www.zzz
14:08:42.318 MTR: =Saved     parameters and configuration
14:08:42.338 MTR: **adding endpoint = 4 type:http_humidity filter:AM2301#Humidity** url:xxx.yyy.www.zzz
14:08:42.759 MTR: =Saved     parameters and configuration
14:08:43.456 MTR: =Saved     parameters and configuration
14:08:46.003 MTR: =Saved     parameters and configuration
14:08:51.558 MTR: =Saved     parameters and configuration
14:08:55.490 CMD: mtrinfo
14:08:55.500 RSL: RESULT = {"MtrInfo":{"Ep":1,"Temperature":5890}}
14:08:55.512 RSL: RESULT = {"MtrInfo":{"Ep":2,"Power":0}}                                      **>>> is missing the AM2301 device**
14:08:55.526 RSL: RESULT = {"MtrInfo":"Done"}

Making a search of matter device with Alexa I see correctly **ALL** the matter devices:
**ESP32-Temp, ESP8266Relay amd ESP8266-Temp** (the humidity is not recognized by Alexa)
In this configuration, **using Alexa-Interface I see the TWO correct temperature and I can send Power-on/off** of the ESP8266-Realy.

This works well also if the MTRINFO command shows **ONLY** 2 devices, Alexa see **ALL** three devices !!!

But If I **reset** the configuration of ESP32 and define a **Relay and the Temp-Sensor**, when I add the ESP8266 with Relay and AM2301, Alexa show **all** four devices but the **none of Temp-Devices shows the temperature**

In this situation the MTRINFO command is similar of before: it shows all the ESP32 devices (Relay and Temp) and ONLY the ESP8266 Relay (in this configuration the ES8266 TempDevice is not shown accordling with the previous configuration)

So the problems are two:

1. The MTRINFO command doesn't show the temp-device on bridged devices but indeed Alexa discovers them
2. Alexa works fine with both the TempDevices (show temperature of ESP32-Temp and ESP8266-Temp) if there is ONLY the TempDevice on ESP32. If I don't configure the Relay on the ESP32 all the TempDevices are shown but I don't see the temperatures values. Sometimes it shows the ESP32-Temp values, sometimes it shows the ESP8266-Temp values, sometimes it shows NONE of the Temp values.

This is the result of "weblog 3" enabled:
15:50:01.450 MTR: HTTP async request 'http://xxx.yyy.www.zzz:80/cm?cmnd=Status+8'
15:50:01.959 MTR: HTTP async-resp in 501 ms from xxx.yyy.www.zzz: [136] '{"StatusSNS":{"Time":"2023-12-...'
15:50:01.983 MTR: HTTP async request 'http://xxx.yyy.www.zzz:80/cm?cmnd=Status+11'
15:50:02.427 MTR: HTTP async-resp in 437 ms from xxx.yyy.www.zzz: [323] '{"StatusSTS":{"Time":"2023-12-...'
15:50:05.211 MTR: HTTP async request 'http://xxx.yyy.www.zzz:80/cm?cmnd=Status+11'
15:50:05.873 MTR: HTTP async-resp in 653 ms from xxx.yyy.www.zzz: [323] '{"StatusSTS":{"Time":"2023-12-...'
15:50:06.495 MTR: HTTP async request 'http://xxx.yyy.www.zzz:80/cm?cmnd=Status+8'
15:50:06.956 MTR: HTTP async-resp in 454 ms from xxx.yyy.www.zzz: [136] '{"StatusSNS":{"Time":"2023-12-...'
15:50:08.246 MTR: HTTP async request 'http://xxx.yyy.www.zzz:80/cm?cmnd=Status+11'
15:50:08.768 MTR: HTTP async-resp in 511 ms from xxx.yyy.www.zzz: [323] '{"StatusSTS":{"Time":"2023-12-...'
15:50:11.478 MTR: HTTP async request 'http://xxx.yyy.www.zzz:80/cm?cmnd=Status+11'
15:50:11.496 MTR: HTTP async request 'http://xxx.yyy.www.zzz:80/cm?cmnd=Status+8'
15:50:11.940 MTR: HTTP async-resp in 437 ms from xxx.yyy.www.zzz: [136] '{"StatusSNS":{"Time":"2023-12-...'
15:50:14.733 MTR: HTTP async request 'http://xxx.yyy.www.zzz:80/cm?cmnd=Status+11'
15:50:15.189 MTR: HTTP async-resp in 451 ms from xxx.yyy.www.zzz: [323] '{"StatusSTS":{"Time":"2023-12-...'
15:50:16.731 MTR: HTTP async request 'http://xxx.yyy.www.zzz:80/cm?cmnd=Status+8'
15:50:17.191 MTR: HTTP async-resp in 452 ms from xxx.yyy.www.zzz: [136] '{"StatusSNS":{"Time":"2023-12-...'
15:50:17.968 MTR: HTTP async request 'http://xxx.yyy.www.zzz:80/cm?cmnd=Status+11'
15:50:18.499 MTR: HTTP async-resp in 524 ms from xxx.yyy.www.zzz: [323] '{"StatusSTS":{"Time":"2023-12-...'
15:50:21.231 MTR: HTTP async request 'http://xxx.yyy.www.zzz:80/cm?cmnd=Status+11'
15:50:21.729 MTR: HTTP async-resp in 490 ms from xxx.yyy.www.zzz: [323] '{"StatusSTS":{"Time":"2023-12-...'
15:50:21.755 MTR: HTTP async request 'http://xxx.yyy.www.zzz:80/cm?cmnd=Status+8'
15:50:22.250 MTR: HTTP async-resp in 488 ms from xxx.yyy.www.zzz: [136] '{"StatusSNS":{"Time":"2023-12-...'
15:50:24.471 MTR: HTTP async request 'http://xxx.yyy.www.zzz:80/cm?cmnd=Status+11'
15:50:24.828 MTR: HTTP async-resp in 349 ms from xxx.yyy.www.zzz: [323] '{"StatusSTS":{"Time":"2023-12-...'

**Reply by s-hadinger** [reply]
The documentation was not clear about it, but `MtrInfo` only works for local sensors. It does not report bridged sensors. This may change since it's helpful to also report values.

For Alexa, I'm unsure how it behaves. Can you provide logs when pairing Alexa ?
